### DISCUSSION TITLE
Error while running the TransducerLoss module

### DISCUSSION THREAD

**Discussion by HizRadit07** [speechbrain/speechbrain#1867] [Q&A] [answered]
Hello, I'm new to speechbrain. I am currently following a tutorial on transducers. Currently, I am facing the following error while using the TransducerLoss module to compute loss function.

`ValueError: Found inputs tensors to be on [device(type='cuda', index=0), device(type='cuda', index=0), device(type='cuda', index=0), device(type='cuda', index=0)] while needed to be on a 'cuda' device to use the transducer loss.`

Looking at the source code, in the transducer_loss.py file, the `forward` function seems to be doing the following comparison.
`if logits.device == labels.device == T.device == U.device == "cuda":`

I was wondering how can I solve this problem? any help would be much appreciated

**Comment by asumagic** [comment, marked as answer]
Sorry for the delay, it is a bug in SpeechBrain. I opened an issue: https://github.com/speechbrain/speechbrain/issues/1928

As a hack, you can edit your local SpeechBrain install to remove the check (just replace it with an `if True:` and make sure your tensors are on the CUDA device yourself).

**Reply by Adel-Moumen** [reply]
PR is merged.
